bin/plot_truncated_pexecs

```diff
@@ -0,0 +1,221 @@
+# to the UPL must be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import argparse
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as pyplot
+from matplotlib.backends.backend_pdf import PdfPages
+import os.path
+import random
+import sys
+
+from statsmodels.stats.proportion import proportion_confint
+
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
+from warmup.krun_results import parse_krun_file_with_changepoints
+from warmup.plotting import add_margin_to_axes, compute_grid_offsets, style_axis, STYLE_DICT
+
+# Set matplotlib styles, similar to Seaborn 'whitegrid'.
+for style in STYLE_DICT:
+    matplotlib.rcParams[style] = STYLE_DICT[style]
+
+NUMBER_TRIALS = 1000
+ALPHA = 0.01  # Significance level.
+METHOD = 'wilson'
+MIN_PEXECS = 2
+MAX_PEXECS = 29
+ORIGINAL_PEXECS = 30
+
+CATEGORIES = ['warmup', 'slowdown', 'flat', 'no steady state']
+
+# Default (PDF) font sizes
+TICK_FONTSIZE = 8
+AXIS_FONTSIZE = 8
+
+GRID_MAJOR_X_DIVS = 10
+GRID_MAJOR_Y_DIVS = 10
+
+LINE_COLOUR = 'k'
+LINE_WIDTH = 1
+
+
+def parse_json(json_file):
+    """Return only classifications from original file."""
+
+    data = None
+    _, data = parse_krun_file_with_changepoints([json_file])
+    assert data is not None, 'No original results file.'
+    assert len(data.keys()) == 1, 'Expected one machine per results file.'
+    machine = data.keys()[0]
+    return data[machine]['classifications']
+
+
+def all_same_category(classifications):
+    """Return True if all classifications fell into one category."""
+
+    total = len(classifications)
+    for category in CATEGORIES:
+        if classifications.count(category) == total:
+            return True
+    return False
+
+
+def generate_trials(json_file):
+    """Generate NUMBER_TRIALS Bernoulli trials for classifications.
+    Ignore <VM, benchmark> pairs with consistent classifications.
+    """
+
+    classifications = parse_json(json_file)  # Original data.
+    rangen = random.Random()
+    trials = dict()
+    for n_pexecs in xrange(MIN_PEXECS, MAX_PEXECS + 1):
+        trials[n_pexecs] = dict()
+        for key in classifications:
+            if not all_same_category(classifications[key]):
+                trials[n_pexecs][key] = list()
+                for _ in xrange(NUMBER_TRIALS):
+                    counts = [0, 0, 0, 0]
+                    for _ in xrange(n_pexecs):
+                        next_class = rangen.choice(classifications[key])
```

Comment:
I'm not saying it will help :) What I'm saying is that we have to know if the method we're using requires sampling with or without replacement, as they have different effects.